llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -399,31 +412,46 @@ auto toHazardType(bool Conflict) {
 // recognizing alternatives
 ScheduleHazardRecognizer::HazardType
 AIEHazardRecognizer::getHazardType(const MCInstrDesc &Desc,
+                                   MemoryBankBits MemoryBanks,
                                    const int DeltaCycles) {
-  return toHazardType(checkConflict(
-      Scoreboard, ItinData, Desc.getSchedClass(),
+  return getHazardType(
+      Desc.getSchedClass(),
       getSlotSet(Desc, *TII->getFormatInterface(), IgnoreUnknownSlotSets),
-      DeltaCycles, FUDepthLimit));
+      MemoryBanks, DeltaCycles);
```

Comment:
Where did FUDepthLimit go?

Reply:
With the above change we will call , and thus `FUDepthLimit` is abstracted there.  

```
ScheduleHazardRecognizer::HazardType
AIEHazardRecognizer::getHazardType(unsigned SchedClass, SlotBits SlotSet,
                                   MemoryBankBits MemoryBanks,
                                   int DeltaCycles) {
  return toHazardType(checkConflict(
      Scoreboard, ItinData, SchedClass, SlotSet, MemoryBanks,
      TII->getMemoryCycles(SchedClass), DeltaCycles, FUDepthLimit));
}
```